client/src/pages/UserPage/UserPage.js

```diff
@@ -23,32 +24,34 @@ const preloaderSX = {
   color: 'secondary',
 };
 
+// TODO: why theme is not working on typography
+
 const UserPage = () => {
-  const [userData, setUserData] = useState(null);
+  const { userData, error, isLoading } = useSelector((store) => store.userReducer);
+  const dispatch = useDispatch();
 
   useEffect(() => {
-    const getUserData = async () => {
-      const res = await getCustomer();
-      setUserData(res);
-    };
-    getUserData();
+    dispatch(fetchUserInfo());
   }, []);
 
   return (
-    <>
-      {userData ? (
-        <Container component={Paper} sx={containerSX}>
+    <Container component={Paper} sx={containerSX}>
+      {isLoading && <CircularProgress sx={preloaderSX} />}
+      {userData && (
+        <>
           <UserHeader userData={userData} />
           <UserList userData={userData} />
           <ButtonContainer />
-        </Container>
-      ) : (
-        <Container component={Paper} sx={containerSX}>
-          <CircularProgress sx={preloaderSX} />
+        </>
+      )}
+      {error && (
+        <Container>
+          <Typography component="h2">Error code: {error.status}</Typography>
+          <Typography component="h3">{error.statusText}</Typography>
+          <Typography component="p">Please try again later</Typography>
```

Comment:
`Typography` is `p` by default
```suggestion
          <Typography>Please try again later</Typography>
```